e2e/test/Helpers/HostNameHelper.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.Devices.E2ETests.Helpers
+{
+    internal static class HostNameHelper
+    {
+        private static readonly Regex s_hostNameRegex = new Regex("(?<=HostName=).*?(?=;)", RegexOptions.Compiled);
```

Comment:
If you are interested in knowing more about this regex, it employs a negative look behind to require, but not match on, some text ahead of the match we want. Then it also uses a negative look ahead to know when to stop the match.

This way you get exactly what you want, and don't have to parse various matches and index on the one you want.

Also note, I used the non-greedy matcher because there are multiple semi-colons in the connection string, so you want to match to the smallest unit to just get the host name.

Reply:
Why don't we just store the IotHubHostName in an Env variable? We do that for Far away hub. That way we don't need to do the regex parsing.

Reply:
Actually, now that you mention it, I'd rather get rid of the far hub hostname and just parse the connection string.

Reply:
Don't forget to update keyvault appropriately if you do. We only really need the hostname for our e2e tests, but I'm not opposed to this change